flake.nix

```diff
@@ -0,0 +1,35 @@
+{
+  description = "wasm-pack setup";
+
+  inputs = {
+    nixpkgs = { url = "github:nixos/nixpkgs/nixos-unstable"; };
+    rust-overlay = { url = "github:oxalica/rust-overlay"; };
+  };
+
+  outputs = { nixpkgs, rust-overlay, ... }:
+    let system = "x86_64-linux";
+    in {
+      devShell.${system} = let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ rust-overlay.overlays.default ];
+        };
+      in (({ pkgs, ... }:
+        pkgs.mkShell {
+          buildInputs = with pkgs; [
+            cargo
+            cargo-watch
+            nodejs
+            wasm-pack
```

Comment:
I am not sure if installing `wasm-pack` is needed here, since I added `wasm-pack` as a npm/yarn dependency and the build is calling `npx wasm-pack build --release` explicitly and not `wasm-pack`.

We could also change it to just use `wasm-pack` and add it as a build requirement and update the GitHub workflows to install `wasm-pack`?

It was just easier to make `wasm-pack` not a build dependency and just let yarn install it.

Reply:
Yeah the flake I found and used had a little more than necessary... and I didn't think about it too hard, since it's probably only going to matter to me. Definitely better not to install things I don't need though (especially a second nodejs; `shell.nix` is setting up Node 20, the flake was *meant* to just be for Rust).